### DISCUSSION TITLE
Understanding Collection

### DISCUSSION THREAD

**Discussion by rostero1** [ariakit/ariakit#3291] [Questions] [answered]
Dumb question.  I'm trying to understand the need for Collection, which is used to track a collection of DOM elements in the order that they're rendered.  I'm not sure why this is needed if we know what components we are rendering.  And is this need something specific to React?

**Comment by diegohaz** [comment, marked as answer]
That's because elements can be rendered asynchronously and out of sequence in React when using compound components. Collection is an abstract component particularly used by other composite components such as Menu, Select, and so on. If you don't grasp its necessity, don't worry, you'll likely never need to directly use Collection unless you're working with lower-level compound component abstractions, at which point you'll fully comprehend the problem.

**Reply by rostero1** [reply]
Perfect
